src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs

```diff
@@ -361,7 +361,7 @@ public void RemoveAppLanguages(IEnumerable<(Guid appId, string languageShortName
                             .Select(aaot => aaot.Agreement)
                             .Select(agreement => new AgreementAssignedOfferData(
                                 agreement!.Id,
-                                agreement.Name,
+                                agreement.AgreementDescriptions.SingleOrDefault(x => x.LanguageShortName == languageShortName)!.Description,
```

Comment:
using `!` on `SingleOrDefault`could lead to an unhandled exception if the result is null which would be the case if one queries the database with an language short name that isn't existing.

my suggestion would be to select all agreementDescriptions with the matching languageShortName and handle the check if only 1 agreement description is set in the business logic.

```suggestion
                                agreement.AgreementDescriptions.Where(x => x.LanguageShortName == languageShortName).Select(ad => ad.Description),
```

Reply:
Thanks for the feedback, I have added a test with non existing language short name and I have received NULL as agreementName... please let me know if this change is still required, quite feasible suggestion though.

Reply:
Oh that's quite nice, thought it would result in an error. consider the comment never been here ;)